File: nibonibo-backend/src/main/java/com/nibonibo/backend/entity/Video.java
package com.nibonibo.backend.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("video")
public class Video {

    @TableId
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private String coverUrl;
    private String videoUrl;
    private Integer duration;
    private Long categoryId;
    private Integer status;
    private Long viewCount;
    private Long likeCount;
    private Long favoriteCount;
    private Long commentCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}

File: nibonibo-frontend/src/types/video.ts
import type { User } from './user'

export interface Video {
  id: string
  title: string
  description: string
  coverUrl: string
  videoUrl: string
  up: Pick<User, 'id' | 'nickname' | 'avatar'>
  playCount: number
  likeCount: number
  favoriteCount: number
  coinCount: number
  tags: string[]
  duration: string
  publishedAt: string
}
